qiskit/transpiler/target.py

```diff
@@ -159,6 +159,14 @@ class Target(BaseTarget):
     angle :class:`~qiskit.circuit.library.RXGate` while ``rx`` will get the
     parameterized :class:`~qiskit.circuit.library.RXGate`.
 
+    .. note::
+        This class is designed to be strictly additive and can be queried via
+        indexing, using the instruction's name as a key. If a user desired to
+        modify any property for an instruction, they can do so via
+        :meth:`.Target.update_instruction_properties` as it is the primary option
+        for doing so. However, modification via indexing is not suppported and
+        doing so might invalidate some of the mappings contained by this class.
```

Comment:
You shouldn't use the passive voice here, it's also a bit wordy, something like this is clearer:

```suggestion
        the mapping protocol, using the instruction's name as a key. You can modify any 
        property for an instruction via the :meth:`.Target.update_instruction_properties` method.
        Modification via the mapping protocol or mutating the attributes of a :class:`.InstructionProperties`
        object is **not** supported and doing so will invalidate the internal state of the object.
```